presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -368,17 +430,49 @@ public StoragePageSink createStoragePageSink(
         return new OrcStoragePageSink(orcDataEnvironment.getFileSystem(fileSystemContext), transactionId, columnIds, columnTypes, bucketNumber);
     }
 
-    private ShardRewriter createShardRewriter(FileSystem fileSystem, long transactionId, OptionalInt bucketNumber, UUID shardUuid, Map<String, Type> columns)
+    ShardRewriter createShardRewriter(
+            FileSystemContext fileSystemContext,
+            FileSystem fileSystem,
+            long transactionId,
+            OptionalInt bucketNumber,
+            UUID shardUuid,
+            int shardRowCount,
+            Optional<UUID> deltaShardUuid,
+            boolean tableSupportsDeltaDelete,
+            Map<String, Type> columns)
     {
-        return rowsToDelete -> {
-            if (rowsToDelete.isEmpty()) {
-                return completedFuture(ImmutableList.of());
-            }
-            return supplyAsync(() -> rewriteShard(fileSystem, transactionId, bucketNumber, shardUuid, columns, rowsToDelete), deletionExecutor);
-        };
+        if (tableSupportsDeltaDelete) {
+            return new DeltaShardRewriter(
+                    shardUuid,
+                    shardRowCount,
+                    deltaShardUuid,
+                    deletionExecutor,
+                    uuid -> this.getRowsFromUuid(fileSystem, uuid),
```

Comment:
Why don't we just pass in `fileSystemContext`, `transactionId`, etc into `DeltaShardRewriter`? Having these two functions is really hard to understand from the interface.

Reply:
> DeltaShardRewriter
Passing in ``fileSystemContext``, ``transactionId``, etc into ``DeltaShardRewriter`` was the original code
Then in the review, @jessesleeping suggested that we shouldn't have those in ``DeltaShardRewriter``, we don't want ``DeltaShardRewriter`` to be overweight and it shouldn't have those variable just to execute some function.